lib/spack/spack/config.py

```diff
@@ -408,28 +408,22 @@ def __init__(self, *scopes):
     @_config_mutator
     def push_scope(self, scope):
         """Add a higher precedence scope to the Configuration."""
-        cmd_line_scope = None
-        if self.scopes:
-            highest_precedence_scope = list(self.scopes.values())[-1]
-            if highest_precedence_scope.name == "command_line":
-                # If the command-line scope is present, it should always
-                # be the scope of highest precedence
-                cmd_line_scope = self.pop_scope()
```

Comment:
@tgamblin @scheibelp This was added in 4b2f51d063111d0966404ff7f1eea627f2d286d8 which is part of #9612 There's no unit test failing on removing this behavior (in fact I have tests failing here due to this behavior + the refactor that keeps some of the context manager in sync with the configuration). 

Was there any reason why this was added? Asking because if there is, we need to add an additional test and restore the behavior _outside_ of this class, i.e. in the calling code:
```python
config.push_scope(...)
config.move_as_top_scope("command_line")
```

Reply:
I want the user to be able to use command-line scopes to have the "last-say": I think they will be frustrated if other configuration implicitly overrides config that they set with `-C`. If I failed to test this, then that is my mistake. IMO the behavior is still worthwhile (and worth testing).

Reply:
Sure, my only point is that shouldn't be part of the `Configuration.push` behavior but should be checked at the caller site where we assemble the configuration. Rationale for saying that:
1. Having an API where subsequent `push` and `pop` don't restore the initial state is very counter-intuitive.
2. `Configuration` should not bother with the names of the scopes, it's code that uses that class that could/should give meaning to them